src/main/java/org/springframework/integration/aws/inbound/kinesis/KinesisMessageDrivenChannelAdapter.java

```diff
@@ -172,6 +173,9 @@
 
 	private ApplicationEventPublisher applicationEventPublisher;
 
+	@Nullable
+	private Function<List<Shard>, List<Shard>> shardsToConsumeMapper;
```

Comment:
May we call it `shardsFilter` instead? 
It is really not mapper which means like this "get something and return something else".
Meanwhile the filter meaning should be straight forwarad - "take this and remove something you don't like"

Reply:
The name is fine by me... but then should we use a (Bi)Predicate instead of a `Function`, as you suggested before?

It probably makes more sense that way, if the semantics really are a filter.

Reply:
Since a simple `Predicate<Shard>` doesn't fit to all the use-case, and `BiPredicate` is too hard to digest, I'd prefer to stick with the `Function<List<Shard>, List<Shard>>`. We probably can call it as `shardListFilter`.
This way it would be replicate a purpose of `FileListFilter` abstraction.

Reply:
Hope fully this is done now